### PR TITLE
feat: camper selection frontend

### PR DESCRIPTION
**Pull request by derrick-yyang**
## Notion ticket link

[Cancellation-Refund-Camper-Selection-Page](https://www.notion.so/uwblueprintexecs/Cancellation-Refund-Camper-Selection-Page-All-states-9740991233214020bea547b8ae12294e)

## Implementation description
* Implemented the Camper Refund/Cancellation page
* Interacts with backend to pull data with respect to the `refundCode` in the URL and displays the available refund options
**Quick Note:** PR is not finished, still need PM or designer opinion on grouping sessions per name.

## Steps to test
1. Run the app locally and go to `localhost:3000/refund/<any-string>`. The string after refund can be anything for now, the backend will accept any string and will return the same test data
2. Check the validity of the states in each card
3. Check that the cost and numbers are correct
4. Check mobile screen sizes and verify it's what we want according to the figma

## What should reviewers focus on?
* Any styling requests?
* Double check the coding style/design to see if it's reusable/robust enough. I occasionally get errors due to the response from our Backend not being the format our frontend code expects so some functions fail
* **_As per the quick note above, would live an external opinion on grouping sessions under the same name. I feel some more decisions need to be made about this, like: "should all sessions under the same name be refunded all at once or not at all? Or have the option to individually be refunded?", "Do we need to change the formating of camper refund cards?" etc._**

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by NathanSouphanthong — APPROVED** [review]
LGTM

File: frontend/src/components/pages/CamperRefundCancellation/CamperRefundInfoCard.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        dates.push(new Date(date));
+      });
+    });
+
+    // sort the dates and find the first session
+    dates.sort((a, b) => a.getTime() - b.getTime());
+    const firstDate = dates[0];
+    const thirtyDays = 30 * 24 * 60 * 60 * 1000;
+
+    return new Date().getTime() - firstDate.getTime() >= thirtyDays;
```
Comment (braydenatnimble) [inline frontend/src/components/pages/CamperRefundCancellation/CamperRefundInfoCard.tsx:93]: believe we want to switch this to:
`return (firstData.getTime() - new Date().getTime()) >= thirtyDays`

The current implementation displays both cards as invalid refunds even though the dates are > 30 days away.

Comment (braydenatnimble) [inline frontend/src/components/pages/CamperRefundCancellation/CamperRefundInfoCard.tsx:93]: nit: while you're here might as well declare this as a const "TODAY" (or similar)
- less magic numbers the better

File: frontend/src/components/pages/CamperRefundCancellation/CamperRefundFooter.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { Flex, Button } from "@chakra-ui/react";
+
+const CamperRefundFooter = (props: any): React.ReactElement => {
+  return (
```
Comment (braydenatnimble) [inline frontend/src/components/pages/CamperRefundCancellation/CamperRefundFooter.tsx:5]: Formatting issue: 
Current implementation has the button un-centered. My suggestion:
1) Change alignItems="flex-end" to alignItems="center" on the flex box
2) remove right margin (mr="50px") on the button itself

**Review by BraydenRoyston — APPROVED** [review]
Looking good locally with a few minor changes.

Approved once those minor comments are addressed. Good stuff!

**Pull request merged**